server/src/main/java/org/apache/iotdb/db/service/metrics/FileMetrics.java

```diff
@@ -52,15 +64,30 @@ public class FileMetrics implements IMetricSet {
   private long innerUnseqCompactionTempFileNum = 0L;
   private long crossCompactionTempFileNum = 0L;
 
+  private long openFileNumbers = 0L;
+  private String[] getOpenFileNumberCommand;
+
   @Override
   public void bindTo(AbstractMetricService metricService) {
-    metricService.createAutoGauge(
-        Metric.FILE_SIZE.toString(),
-        MetricLevel.IMPORTANT,
-        this,
-        FileMetrics::getWalFileTotalSize,
-        Tag.NAME.toString(),
-        "wal");
+    bindTsFileMetrics(metricService);
+    bindWalFileMetrics(metricService);
+    bindCompactionFileMetrics(metricService);
+    bindSystemRelatedMetrics(metricService);
+    // finally start to update the value of some metrics in async way
+    if (null == currentServiceFuture) {
```

Comment:
Maybe we can remove this async logic now because `collect` is not time-consuming anymore.